### PR TITLE
Upgrade to Rails 7 and mongoid 8

### PR DESCRIPTION
**Pull request by jkotanchik-SB**
The MADiE Team is working towards using [cqm-reports](https://github.com/projecttacoma/cqm-reports) to generate QRDA exports and need to update cqm-models to resolve vulnerabilities in rails 5 and nokogiri.

I opted to move rails up as far as I could to avoid having to do this dance again anytime soon. That resulted in Rails 7 and Mongoid 8, as well as the below changes to resolves unit test failures. I'm not deeply familiar with this codebase, so any feedback on changes made would be appreciated.

If moving to Rails 6 would be preferred, that is an option. I did test with Rails 6 and Mongoid 7, and though I had to make the same changes, it did result in a clean report from `bundle audit`.

If preferred, we can skip merging to `master` and maintain these changes on their own branch. We also considered releasing these changes as new minor version to the ruby gem.

**Submitter:**
- [ ] This pull request describes why these changes were made.
- [ ] Internal ticket for this PR:
- [ ] Internal ticket links to this PR
- [ ] Code diff has been done and been reviewed
- [ ] Tests are included and test edge cases
- [ ] Tests have been run locally and pass
- [ ] If applicable, the library version number in `package.json` and `cqm-models.gemspec` has been updated
- [ ] Cqm-execution fixtures have been updated with the update_cqm_execution_fixtures.sh script inside server-scripts using this branch in the cqm-converter

**Bonnie Reviewer:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Cypress Reviewer:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: app/models/qdm/basetypes/data_element.rb [hunk 1]
```diff
@@ -68,15 +68,22 @@ def shift_dates(seconds)
         send(field + '=', (send(field).to_time + seconds.seconds).to_datetime) if send(field).is_a? Time
         send(field + '=', send(field).shift_dates(seconds)) if (send(field).is_a? Interval) || (send(field).is_a? DataElement)
 
-        # Special case for facility locations
-        if field == 'facilityLocations'
+        # Special case for facility locations and result
+        if field == 'result'
```
Comment (dczulada) [inline app/models/qdm/basetypes/data_element.rb:72]: @jkotanchik-SB the branch for 'result' is not needed.

The code below takes care of shifting the result.  The branch we were using didn't have that line. 

send(field + '=', (send(field).to_time + seconds.seconds).to_datetime) if send(field).is_a? Time

Comment (dczulada) [reply app/models/qdm/basetypes/data_element.rb:72]: I should have realized that we had already taken care of that with this PR.

https://github.com/projecttacoma/cqm-models/pull/195

Comment (jkotanchik-SB) [reply app/models/qdm/basetypes/data_element.rb:72]: Then, can I safely remove this whole branch from the the `if` block?

Comment (dczulada) [reply app/models/qdm/basetypes/data_element.rb:72]: Yes you can

**Review by dczulada — APPROVED** [review]

**Review by adongare — APPROVED** [review]

**Comment by jkotanchik-SB** [comment]
@dczulada Just to be sure you saw, I bumped the ruby version from 4.1.X to 4.2.0. I did not change the version in package.json. 

Does that change make sense for your workflows? And would you like me to bump up the package.json version as well?

**Comment by dczulada** [comment]
> @dczulada Just to be sure you saw, I bumped the ruby version from 4.1.X to 4.2.0. I did not change the version in package.json.
> 
> Does that change make sense for your workflows? And would you like me to bump up the package.json version as well?

The PR here does bump the version to 4.2.0 in the gemspec.  I think that version increment makes sense to me.
